kubernetes/base/watch/watch.py

```diff
@@ -172,14 +172,19 @@ def stream(self, func, *args, **kwargs):
         # We want to ensure we are returning within that timeout.
         disable_retries = ('timeout_seconds' in kwargs)
         retry_after_410 = False
+        deserialize = kwargs.pop('deserialize', True)
         while True:
             resp = func(*args, **kwargs)
             try:
                 for line in iter_resp_lines(resp):
                     # unmarshal when we are receiving events from watch,
                     # return raw string when we are streaming log
                     if watch_arg == "watch":
-                        event = self.unmarshal_event(line, return_type)
+                        if deserialize:
+                            event = self.unmarshal_event(line, return_type)
```

Comment:
We actually have the [return_type flag](https://github.com/kubernetes-client/python/blob/d80165da21a9fec1d5abc785cb446ab16f6affdb/kubernetes/base/watch/watch.py#L87) which could serve the purpose, but it looks pretty unintuitive-- IIUC `_raw_return_type` can be [persisted](https://github.com/kubernetes-client/python/blob/d80165da21a9fec1d5abc785cb446ab16f6affdb/kubernetes/base/watch/watch.py#L87) by setting `return_type` to `False`, which would result in deserialization [being skipped](https://github.com/kubernetes-client/python/blob/d80165da21a9fec1d5abc785cb446ab16f6affdb/kubernetes/base/watch/watch.py#L110-L116)

That being said, the flag name seems unintuitive and I think `deserialize = False` makes more sense.

cc @yliaog

Reply:
+1 on deserialize = False